### PR TITLE
fix bs check server

### PR DESCRIPTION
**Pull request by caoxianfei1**
Signed-off-by: pengpengSir <2968332764@qq.com>

fix bs check server

Signed-off-by: pengpengSir <2968332764@qq.com>

seperate command

Signed-off-by: pengpengSir <2968332764@qq.com>

Update server.go

Signed-off-by: pengpengSir <82749014+pengpengSir@users.noreply.github.com>

[tools-v2] add bs check server

### What problem does this PR solve?

Issue Number: #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by caoxianfei1** [comment]
cicheck

File: tools-v2/internal/error/error.go [hunk 1]
```diff
@@ -469,6 +469,9 @@ var (
 	ErrSnapShotAddrNotConfigured = func() *CmdError {
 		return NewInternalCmdError(70, "get snapshotAddr fail, err: %s")
 	}
+	ErrBsGetChunkServerInfos = func() *CmdError {
+		return NewInternalCmdError(55, "get chunkserver infos fail, err: %s")
```
Comment (montaguelhz) [inline tools-v2/internal/error/error.go:473]: The number is 71 now.

File: tools-v2/pkg/cli/command/curvebs/check/server/server.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+func (sCmd *ServerCommand) Print(cmd *cobra.Command, args []string) error {
+	return output.FinalCmdOutput(&sCmd.FinalCurveCmd, sCmd)
+}
+
+func (sCmd *ServerCommand) ResultPlainOutput() error {
+	return output.FinalCmdOutputPlain(&sCmd.FinalCurveCmd)
+}
+
+func (sCmd *ServerCommand) RunCommand(cmd *cobra.Command, args []string) error {
+	// 这里就可以得到指定的server下的所有的chunkserver
```
Comment (montaguelhz) [inline tools-v2/pkg/cli/command/curvebs/check/server/server.go:103]: Maybe should put comments translated into English.

File: tools-v2/pkg/cli/command/curvebs/check/server/server.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	chunkServerInfos, err := chunkserver.ListChunkServerInfos(sCmd.Cmd)
+	if err.TypeCode() != cmderror.CODE_SUCCESS {
+		return err.ToError()
+	}
+	if len(chunkServerInfos) == 0 {
+		return nil
+	}
+
+	// 需要遍历所有的chunkserver下的copyset,获取copyset的健康情况
```
Comment (montaguelhz) [inline tools-v2/pkg/cli/command/curvebs/check/server/server.go:112]: ditto

File: tools-v2/pkg/cli/command/curvebs/check/server/server.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+	config.AddBsChunkServerIdFlag(sCmd.Cmd)
+	config.AddBsLogicalPoolIdRequiredFlag(sCmd.Cmd)
+	config.AddBsCopysetIdRequiredFlag(sCmd.Cmd)
+	for _, item := range chunkServerInfos {
+		row := make(map[string]string)
+		chunkId := item.GetChunkServerID()
+		// 得到对应的chunkserver下面的所有的copyset info
```
Comment (montaguelhz) [inline tools-v2/pkg/cli/command/curvebs/check/server/server.go:119]: ditto

File: tools-v2/pkg/cli/command/curvebs/check/server/server.go [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+		total := len(copySets)
+		var unhealthy int
+		for _, copyset := range copySets {
+			logicalPoolId := copyset.GetLogicalPoolId()
+			copysetId := copyset.GetCopysetId()
+
+			copysetKey := cobrautil.GetCopysetKey(uint64(logicalPoolId), uint64(copysetId))
+			sCmd.Cmd.Flags().Set(config.CURVEBS_COPYSET_ID, strconv.FormatUint(uint64(copysetId), 10))
+			sCmd.Cmd.Flags().Set(config.CURVEBS_LOGIC_POOL_ID, strconv.FormatUint(uint64(logicalPoolId), 10))
+			// sCmd.Cmd.ParseFlags([]string{
```
Comment (montaguelhz) [inline tools-v2/pkg/cli/command/curvebs/check/server/server.go:135]: remove unused code.

File: tools-v2/pkg/cli/command/curvebs/list/chunkserver/chunkserver.go [hunk 6]
```diff
@@ -107,17 +112,38 @@ func (pCmd *ChunkServerCommand) Init(cmd *cobra.Command, args []string) error {
...
+	serverPort := config.GetBsFlagUint32(pCmd.Cmd, config.CURVEBS_SERVER_PORT)
+
 	servers, err := server.ListServer(pCmd.Cmd)
 	if err.TypeCode() != cmderror.CODE_SUCCESS {
 		return err.ToError()
 	}
+
+	// Filter servers by serverId or serverIp
+	if serverId != 0 && serverIp != "" {
+		return fmt.Errorf("flag --%s and -- %s cannot be used together",
```
Comment (montaguelhz) [inline tools-v2/pkg/cli/command/curvebs/list/chunkserver/chunkserver.go:126]: There's an extra space.

**Review by montaguelhz — CHANGES_REQUESTED** [review]
good job

File: tools-v2/internal/error/error.go (hunk 1 above)
Comment (caoxianfei1) [reply tools-v2/internal/error/error.go:473]: fixed

**Pull request closed without merging**